code/modules/vehicles/mecha/mecha_defense.dm

```diff
@@ -0,0 +1,324 @@
+/*!
+ * # Mecha defence explanation
+ * Mechs focus is on a more heavy-but-slower damage approach
+ * For this they have the following mechanics
+ *
+ * ## Backstab
+ * Basically the tldr is that mechs are less flexible so we encourage good positioning, pretty simple
+ * ## Armor modules
+ * Pretty simple, adds armor, you can choose against what
+ * ## Internal damage
+ * When taking damage will force you to take some time to repair, encourages improvising in a fight
+ * Targetting different def zones will damage them to encurage a more strategic approach to fights
+ * where they target the "dangerous" modules
+ */
+
+/// tries to damage mech equipment depending on damage and where is being targetted
+/obj/vehicle/sealed/mecha/proc/try_damage_component(damage, def_zone)
+	if(damage < component_damage_threshold)
+		return
+	var/obj/item/mecha_parts/mecha_equipment/gear
+	switch(def_zone)
+		if(BODY_ZONE_L_ARM)
+			gear = equip_by_category[MECHA_L_ARM]
+		if(BODY_ZONE_R_ARM)
+			gear = equip_by_category[MECHA_R_ARM]
+	if(!gear)
+		return
+	var/brokenstatus = gear.obj_integrity
+	// always leave at least 1 health
+	brokenstatus--
+	var/damage_to_deal = min(brokenstatus, damage)
+	if(!damage_to_deal)
+		return
+	gear.take_damage(damage_to_deal)
```

Comment:
that's a weird way of doing lel, try to fix it

Reply:
fixed